exchanges/auth/src/authExchange.ts

```diff
@@ -233,10 +234,17 @@ export function authExchange(
                 result$,
                 onStart(() => {
                   const operation = addAuthToOperation(baseOperation);
-                  bypassQueue.add(operation);
+                  bypassQueue.add(
+                    operation.context._instance as OperationInstance
+                  );
                   retries.next(operation);
                 }),
-                filter(result => result.operation.key === baseOperation.key),
+                filter(
+                  result =>
+                    result.operation.key === baseOperation.key &&
+                    baseOperation.context._instance ===
+                      result.operation.context._instance
```

Comment:
I'm pretty sure that this can become `0` but we could probably just do the `.has` check either way and allow `undefined` on the `Set` type since we can just make sure not to add `undefined` to the set lol